tsconfig.json

```diff
@@ -0,0 +1,66 @@
+{
+  // @see https://www.typescriptlang.org/tsconfig
+  // @see https://www.typescriptlang.org/docs/handbook/project-references.html
+  // @see https://www.typescriptlang.org/docs/handbook/configuring-watch.html
+  // when conflicted, follow https://bun.sh/docs/ecosystem/typescript#compiler-options
+
+  "$schema": "https://json.schemastore.org/tsconfig",
+  "watchOptions": {
+    "excludeDirectories": [
+      "**/node_modules",
+      "**/.git",
+      "**/build",
+      "**/.turbo",
+      "**/dist",
+      "**/lib"
+    ]
+  },
+  "exclude": [
+    "**/node_modules",
+    "**/.git",
+    "**/build",
+    "**/.turbo",
+    "**/dist",
+    "**/lib"
+  ],
+  "compilerOptions": {
+    "allowJs": true,
+    "allowSyntheticDefaultImports": true,
+    "assumeChangesOnlyAffectDirectDependencies": true,
+    "baseUrl": ".",
+    "checkJs": false, // throws too many errors in third party shit
+    "composite": true,
+    "declaration": true,
+    "declarationMap": true,
+    "emitDecoratorMetadata": true,
+    "esModuleInterop": true,
+    "experimentalDecorators": true,
+    "forceConsistentCasingInFileNames": true,
+    "incremental": true,
+    "isolatedModules": false,
+    "jsx": "react-jsx",
+    "lib": ["esnext"],
+    "module": "esnext",
+    "moduleResolution": "bundler",
+    "noEmit": false,
+    "noErrorTruncation": false,
+    "noFallthroughCasesInSwitch": false,
+    "noImplicitAny": false, // never set this to false
```

Comment:
lol

Reply:
i think we should set this back to true and force use of `ts-ignore` or something like that